### DISCUSSION TITLE
Installation of Sunbird-Lern User&Org service

### DISCUSSION THREAD

**Discussion by devaj04** [Sunbird-Lern/Community#95] [Installation] [answered]
Trying to get health API - http://localhost:9000/health
In response getting, **Elasticsearch CONNECTION_ERROR**

**I have followed the below steps still facing the issue.**

<img width="959" alt="image" src="https://user-images.githubusercontent.com/125238862/227190302-cf83648d-70ea-4ece-a42d-d5e1b86bb453.png">

{
"id": "api.all.health",
"ver": "health",
"ts": "2023-03-23 15:11:09:408+0530",
"params": {
"resmsgid": "2bbb0165-bfb7-4361-9228-0cd8473d489b",
"msgid": "2bbb0165-bfb7-4361-9228-0cd8473d489b",
"err": null,
"status": "SUCCESS",
"errmsg": null
},
"responseCode": "OK",
"result": {
"response": {
"checks": [
{
"err": "",
"healthy": true,
"name": "Learner service",
"errmsg": ""
},
{
"err": "",
"healthy": true,
"name": "Actor service",
"errmsg": ""
},
{
"err": "",
"healthy": true,
"name": "Cassandra service",
"errmsg": ""
},
{
"err": "CONNECTION_ERROR",
"healthy": false,
"name": "Elastic search service",
"errmsg": "Connection error"
}
],
"healthy": false,
"name": "Complete health check api"
}
}
}

**Comment by BharathwajShankar** [comment]
@devaj04 Kindly please check if the elastic search docker container is up and running on your local machine. Also please check all elastic search-related environment variables got settled up correctly in your local machine. If you still face the issue, run the application in the debug mode and keep breakpoints in the initialiseRestClientConnection() method of ConnectionManager.java and see whether the values of cluster, hostname and port get set up correctly.

**Reply by devaj04** [reply]
@BharathwajShankar 
Could you please help with this 
I am doing Cassandra migration, and follow the below step and getting error
https://lern.sunbird.org/use/developer-installation/user-and-org-service/installation-guide

<img width="958" alt="image" src="https://user-images.githubusercontent.com/125238862/231660767-840d020d-2f9e-448c-b369-051907bc28be.png">

**Comment by BharathwajShankar** [comment]
@devaj04 

1. Kindly checkout to release-5.2.0 branch in sunbird-utils
2. Go to the path <project base path>/sunbird-utils/sunbird-cassandra-migration/cassandra-migration and then run,
java -jar 
-Dcassandra.migration.scripts.locations=filesystem:<your project baspath>/sunbird-utils/sunbird-cassandra-migration/cassandra-migration/db/migration/cassandra 
-Dcassandra.migration.cluster.contactpoints=localhost 
-Dcassandra.migration.cluster.port=9042 
-Dcassandra.migration.cluster.username=username 
-Dcassandra.migration.cluster.password=password 
-Dcassandra.migration.keyspace.name=keyspace_name 
target/*-jar-with-dependencies.jar migrate

(you have to replace respective keyspace,filepath in the above command)

**Reply by devaj04** [reply]
@BharathwajShankar now got another issue
java -jar -Dcassandra.migration.scripts.locations=D:/Workspace/Sunbird-Lern/UserOrg-Service/sunbird-utils/sunbird-cassandra-migration/cassandra-migration/src/main/resources/db/migration/cassandra -Dcassandra.migration.cluster.contactpoints=localhost -Dcassandra.migration.cluster.port=9042 -Dcassandra.migration.cluster.username=username -Dcassandra.migration.cluster.password=password -Dcassandra.migration.keyspace.name=sunbird cassandra-migration-0.0.1-SNAPSHOT.jar migrate

no main manifest attribute, in cassandra-migration-0.0.1-SNAPSHOT.jar

**Reply by BharathwajShankar** [reply]
@devaj04 please keep this as it it is and don't change it. I think that should fix this issue.

 target/*-jar-with-dependencies.jar migrate

**Reply by devaj04** [reply]
@BharathwajShankar 
<img width="956" alt="image" src="https://user-images.githubusercontent.com/125238862/231952482-8ac7d6e6-da5b-4211-a12d-ce5bde8bd65a.png">

**Comment by BharathwajShankar** [comment]
@devaj04 please run mvn clean install -DskipTests before running the migration command

**Reply by devaj04** [reply]
@BharathwajShankar 
<img width="960" alt="image" src="https://user-images.githubusercontent.com/125238862/231962737-2f950f49-2c68-4cd8-98b4-5366b161c4a8.png">
<img width="956" alt="image" src="https://user-images.githubusercontent.com/125238862/231962832-bdcd113d-e39d-4240-a15d-2e8887b77d96.png">

**Reply by AmiableAnil** [reply, marked as answer]
@devaj04 
Please follow the following steps:

1. Go to **cassandra-migration**
2. Run **mvn clean install -DskipTests**
3. Run `java -jar \
-Dcassandra.migration.scripts.locations=filesystem:src/main/resources/db/migration/cassandra/sunbird \
-Dcassandra.migration.cluster.contactpoints=localhost \
-Dcassandra.migration.cluster.port=9042 \
-Dcassandra.migration.cluster.username=<username>\
-Dcassandra.migration.cluster.password=<password>\
-Dcassandra.migration.keyspace.name=sunbird \
target/*-jar-with-dependencies.jar migrate`

cc: @BharathwajShankar @reshmi-nair 
**Note:** set your cassandra username and password.

**Comment by devaj04** [comment]
@BharathwajShankar 

After successful of the Cassandra migration, now i am trying to check the Health API of the User and Org service following the below link
https://lern.sunbird.org/use/developer-installation/user-and-org-service/installation-guide

and getting issues in Elasticsearch

{
    "id": "api.all.health",
    "ver": "health",
    "ts": "2023-04-17 14:57:47:554+0530",
    "params": {
        "resmsgid": "d2ed5903-42c4-4764-8b97-aa1610d538f7",
        "msgid": "d2ed5903-42c4-4764-8b97-aa1610d538f7",
        "err": null,
        "status": "SUCCESS",
        "errmsg": null
    },
    "responseCode": "OK",
    "result": {
        "response": {
            "checks": [
                {
                    "err": "",
                    "healthy": true,
                    "name": "Learner service",
                    "errmsg": ""
                },
                {
                    "err": "",
                    "healthy": true,
                    "name": "Actor service",
                    "errmsg": ""
                },
                {
                    "err": "",
                    "healthy": true,
                    "name": "Cassandra service",
                    "errmsg": ""
                },
                {
                    "err": "CONNECTION_ERROR",
                    "healthy": false,
                    "name": "Elastic search service",
                    "errmsg": "Connection error"
                }
            ],
            "healthy": false,
            "name": "Complete health check api"
        }
    }
}

<img width="947" alt="image" src="https://user-images.githubusercontent.com/125238862/232443827-44bfedec-566b-4b00-8b80-abb2a7ec8de3.png">

**Reply by BharathwajShankar** [reply]
@devaj04 please make sure the environment variables are settled properly for elastic search-related configurations in bashrc file and environment variables.

export sunbird_es_host=localhost 
export sunbird_es_port=9200
export sunbird_es_cluster=docker-cluster  **//replace with your cluster**
export user_index_alias=user
export org_index_alias=org

**Comment by devaj04** [comment]
Thank you now we got elasticsearch healthy true
@AmiableAnil  @BharathwajShankar
